src/main/java/com/shapesecurity/shift/validator/Validator.java

```diff
@@ -96,11 +96,15 @@ private static boolean checkIsValidIdentifierName(String name) {
     return ImmutableList.from(Director.reduceScript(new Validator(), script).errors);
   }
 
+  public static ImmutableList<ValidationError> validate(Module module) {
+    return ImmutableList.from(Director.reduceModule(new Validator(), module).errors);
+  }
+
   private boolean checkIsStringLiteral(String rawValue) {
-    Tokenizer tokenizer ;
+    Tokenizer tokenizer;
     try {
       tokenizer = new Tokenizer("\'"+rawValue+"\'", false);
-      Token token = tokenizer.advance();
+      Token token = tokenizer.lookahead;
```

Comment:
Use `lookahead` below as well. Also, you should make sure the token stream length is 2 (because of EOF token) to prevent strings like `"a' + 'b"` from passing this test.